### PR TITLE
TestFixConnLeak_ISSUE89: Change the way it detects closed conns

### PR DESCRIPTION
**Pull request by tcard**
In the new CI, we often see this:

```
--- FAIL: TestFixConnLeak_ISSUE89 (6.87s)
    rest_client_test.go:574: expected conn 10.1.0.4:47296 to be closed
    rest_client_test.go:574: expected conn 10.1.0.4:47308 to be closed
    rest_client_test.go:574: expected conn 10.1.0.4:47312 to be closed
```

It's probably just a race condition, so, fix that by waiting a bit.

Also, the way it was checking that conns were closed was by trying to
reuse them, seeing if that failed with a particular error message.
That's too implementation-dependent. Now it checks that the Close()
method is called on the connections.

### CODE REVIEW

**Comment by sacOO7** [comment]
If the issue is fixed, CI shouldn't fail, right?

**Comment by tcard** [comment]
> If the issue is fixed, CI shouldn't fail, right?

Right. Now it doesn't :)

File: ably/rest_client_test.go [hunk 1]
```diff
@@ -567,16 +577,21 @@ func TestFixConnLeak_ISSUE89(t *testing.T) {
...
+			return atomic.LoadUintptr(&c.closed) != 0
+		}) {
+			t.Errorf("conn to %v wasn't closed", c.RemoteAddr())
 		}
 	}
 }
 
-func connIsClosed(err error) bool {
-	return strings.Contains(err.Error(), "use of closed network connection")
+type connTrackingClose struct {
```
Comment (sacOO7) [inline ably/rest_client_test.go:589]: Maybe we can rename it to `connCloseTracking` / `connCloseTracker`

Comment (tcard) [reply ably/rest_client_test.go]: Done at 506ab43.

**Pull request merged**